File: app/src/main/java/com/yuta/__old/ui/dialogs/user/LogoutDialog.java
package com.yuta.__old.ui.dialogs.user;

import android.content.Context;
import androidx.fragment.app.Fragment;
import com.yuta.__old.R;
import com.yuta.__old.ui.dialogs.CustomInteractiveDialog;
import com.yuta.common.util.UserUtils;

@SuppressWarnings("ConstantConditions")
public class LogoutDialog extends CustomInteractiveDialog {

    public LogoutDialog(Context context, Fragment fragment) {
        super(context, fragment);
        setDialogLayout(R.layout.dialog_logout);
    }

    @Override
    public void start() {
        super.start();

        dialog.findViewById(R.id.close).setOnClickListener(v -> dismiss());
        dialog.findViewById(R.id.submit).setOnClickListener(v -> {
            UserUtils.logOut(activity);
            dismiss();
        });
    }
}

File: app/src/main/java/com/yuta/authorization/ui/LogoutDialog.kt
package com.yuta.authorization.ui

import android.widget.Button
import androidx.fragment.app.Fragment
import com.yuta.app.R
import com.yuta.common.ui.CancelableDialog
import com.yuta.common.util.UserUtils

class LogoutDialog(private val fragment: Fragment) :
    CancelableDialog(R.layout.dialog_logout, fragment.requireActivity()) {

    override fun start() {
        super.start()

        dialog.findViewById<Button>(R.id.close).setOnClickListener { view -> dismiss() }
        dialog.findViewById<Button>(R.id.submit).setOnClickListener { view ->
            UserUtils.logout(fragment.requireActivity())
            dismiss()
        }
    }
}
